src/app/modules/main/profile_section/profile/module.nim

```diff
@@ -157,6 +162,67 @@ method storeProfileShowcasePreferences(self: Module,
     revealedAddresses
   )
 
+method saveProfileShowcasePreferences*(self: Module, showcase: ProfileShowcaseSaveData) =
```

Comment:
@alexjba @noeliaSD @micieslak 
this part looks dirty from my perspective, to simplify it we need to add some extra data to the initial json: 
- wallet accounts metadata
- collectibles's raw identification fields
-and i expect some extra data for tokens (contractAddress and tokeId)

Reply:
`the initial json` means the json composed in qml when saving?

I find it ok if this data is really needed here, in `saveProfileShowcasePreferences`. My reasoning is that the profileShowcasePreferences only operates on visibility and position. If the metadata is needed on save operation, I find it ok to be collected on save as opposed to be carried around in other places where it's not immediatly needed.

But I don't really understand why this data needs to be collected here just to be forwarded to status-go. I'd expect status-go not to need accounts metadata for example when saving some accounts showcase preferences. IMO this data could be kept in sync by status-go internals.

Unless I'm missing something, I think it would be more or less the same if we're collecting this data in `saveProfileShowcasePreferences` or qml. The best solution IMO is for this data not to be needed on save. The profile showcase preferences is not meant to save accounts metadata, just some visibilities and positions.

Reply:
On the other hand, I see it's quite an overhead to parse the qml updates. The profile settings are going through the following changes:

---qml---
1. model to string
---nim---
2. string -> json
3. json -> nim object 1
4. nim object 1 -> nim object 2 (similar object)
5. nim object 2 -> json
6. json -> back to string

would your proposal help in simplifying this flow as well?

Reply:
@alexjba thanks for the detailed comment, let's try to figure out the best way to do it!

Yes, by json I meant the data that qml passes to nim for saving. 

There is one thing about filling additional fields in status-go: these data are in different services and not all status-go api are available from messenger service. It is still possible to route them, but it is much more laborious than in nim.  However, for profile showcase preferences this problem is not so noticeable. I can move wallet filling to go, and it will even be correct. But not everything is so easy with tokens. The most correct structure seems to me to be:

```
{
  communities: [ { communityId: <string>, visibility: <int> }],
  accounts: [ { address <string>, visibility: <int> }],
  collectibles: [ { chainId: <int>, tokenId: <string>, contractAddress: <string>, visibility: <int> }],
  verifiedTokens: [ { symbol: <string>, visibility: <int> }],
  unverifiedTokens: [ { chainId: <int>, contractAddress: <string>, visibility: <int> }],
  socialLinks: [ { url: <string>, text: <string>, visibility: <int> }]
}
```

I would prefer to use the same nim structure for both saving and querying profile showcase preferences from status-go. And it is possible to completely remove converying in the nim layer, but then the position and correct id names will have to be filled in qml.

Reply:
For `verifiedTokens` and `unverifiedTokens` I'd use the word `assets` instead of `tokens` to have more nomenclature consistency!

Reply:
> @alexjba thanks for the detailed comment, let's try to figure out the best way to do it!
> 
> Yes, by json I meant the data that qml passes to nim for saving.
> 
> There is one thing about filling additional fields in status-go: these data are in different services and not all status-go api are available from messenger service. It is still possible to route them, but it is much more laborious than in nim. However, for profile showcase preferences this problem is not so noticeable. I can move wallet filling to go, and it will even be correct. But not everything is so easy with tokens. The most correct structure seems to me to be:
> 
> ```
> {
>   communities: [ { communityId: <string>, visibility: <int> }],
>   accounts: [ { address <string>, visibility: <int> }],
>   collectibles: [ { chainId: <int>, tokenId: <string>, contractAddress: <string>, visibility: <int> }],
>   verifiedTokens: [ { symbol: <string>, visibility: <int> }],
>   unverifiedTokens: [ { chainId: <int>, contractAddress: <string>, visibility: <int> }],
>   socialLinks: [ { url: <string>, text: <string>, visibility: <int> }]
> }
> ```
> 
> I would prefer to use the same nim structure for both saving and querying profile showcase preferences from status-go. And it is possible to completely remove converying in the nim layer, but then the position and correct id names will have to be filled in qml.

I'm not sure if I'm understanding correctly what you say here but if I'm not wrong for the UI to do the proper joins we will need to indentify each item by a `key` being this `key` the following corresponding data for each case:
- Communities --> `key` = `communityId`
- Accounts --> `key` = `walletAddress`
- VerifiedTokens (assets) --> `key` = `symbol` in case of _non-community_ ones but a `hash(chainId, tokenId)` or something (we need to look at the `assetsModel` from wallet) for _community ones_ 
- UnverifiedTokens --> I really don't know about how these are treat in wallet
- Collectibles --> `key` = `uid` value build from _chainId_, _tokenId_ and _contractAddress_ 

I think that if there's the need of some `key` transformation, it should happen in `nim` and the UI should not know anything about the rest of params, just the `key`.

Reply:
> I think that if there's the need of some key transformation, it should happen in nim and the UI should not know anything about the rest of params, just the key.

I think it's a good point to keep the nim specific logic in nim so that we don't have to duplicate it in qml as well.